src/components/HeaderPageLayout.tsx

```diff
@@ -1,114 +1,108 @@
-import PropTypes from 'prop-types';
 import React, {useMemo} from 'react';
-import {ScrollView, View} from 'react-native';
-import _ from 'underscore';
+import {ScrollView, StyleProp, View, ViewStyle} from 'react-native';
 import useNetwork from '@hooks/useNetwork';
 import useStyleUtils from '@hooks/useStyleUtils';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import useWindowDimensions from '@hooks/useWindowDimensions';
 import * as Browser from '@libs/Browser';
+import ChildrenProps from '@src/types/utils/ChildrenProps';
 import FixedFooter from './FixedFooter';
 import HeaderWithBackButton from './HeaderWithBackButton';
-import headerWithBackButtonPropTypes from './HeaderWithBackButton/headerWithBackButtonPropTypes';
+import HeaderWithBackButtonProps from './HeaderWithBackButton/types';
 import ScreenWrapper from './ScreenWrapper';
 
-const propTypes = {
-    ...headerWithBackButtonPropTypes,
-
-    /** Children to display in the lower half of the page (below the header section w/ an animation) */
-    children: PropTypes.node.isRequired,
-
+type HeaderPageLayoutProps = ChildrenProps & {
     /** The background color to apply in the upper half of the screen. */
-    backgroundColor: PropTypes.string,
+    backgroundColor?: string;
 
     /** A fixed footer to display at the bottom of the page. */
-    footer: PropTypes.node,
+    footer?: React.ReactNode;
 
     /** The image to display in the upper half of the screen. */
-    header: PropTypes.node,
+    headerContent?: React.ReactNode;
 
     /** Style to apply to the header image container */
-    // eslint-disable-next-line react/forbid-prop-types
-    headerContainerStyles: PropTypes.arrayOf(PropTypes.object),
+    headerContainerStyles?: Array<StyleProp<ViewStyle>>;
```

Comment:
```suggestion
    headerContainerStyles?: StyleProp<ViewStyle>;
```

When you look at StyleProp you'll realise why array is not needed:
```ts
type Falsy = undefined | null | false;
interface RecursiveArray<T>
  extends Array<T | ReadonlyArray<T> | RecursiveArray<T>> {}
/** Keep a brand of 'T' so that calls to `StyleSheet.flatten` can take `RegisteredStyle<T>` and return `T`. */
type RegisteredStyle<T> = number & {__registeredStyleBrand: T};
export type StyleProp<T> =
  | T
  | RegisteredStyle<T>
  | RecursiveArray<T | RegisteredStyle<T> | Falsy>
  | Falsy;
```